### PR TITLE
Home airport changes

### PR DESCRIPTION
**Pull request by rohilshahZC**

### CODE REVIEW

File: zoomdls/src/main/java/com/zoomcar/uikit/segmentedcontrol/ZSegmentedControl.kt [hunk 1]
```diff
@@ -96,15 +125,27 @@ class ZSegmentedControl : ConstraintLayout, ZSegmentedControlAdapter.IZSegmented
...
+        var itemCellSize = 0
+        when (model?.type) {
+            SegmentControlType.DEFAULT -> {
+                itemCellSize =
+                    ((recyclerSize - 2 * UiUtil.dpToPixels(4, context)) / (model?.list?.size
+                        ?: 1).toFloat()).toInt()
+            }
+            SegmentControlType.WITH_IMAGE -> {
+                itemCellSize = (recyclerSize / (model?.list?.size ?: 1).toFloat()).toInt()
+            }
```
Comment (shishir5) [inline zoomdls/src/main/java/com/zoomcar/uikit/segmentedcontrol/ZSegmentedControl.kt:137]: inline this

File: zoomdls/src/main/java/com/zoomcar/uikit/segmentedcontrol/ZSegmentedControl.kt [hunk 2]
```diff
@@ -115,4 +156,9 @@ class ZSegmentedControl : ConstraintLayout, ZSegmentedControlAdapter.IZSegmented
             listener?.onSegmentSelected(data.id, data.list[position].id)
         }
     }
+
+    enum class SegmentControlType {
+        DEFAULT,
+        WITH_IMAGE;
```
Comment (shishir5) [inline zoomdls/src/main/java/com/zoomcar/uikit/segmentedcontrol/ZSegmentedControl.kt:162]: can we name this LABEL & ICON_LABEL

File: zoomdls/src/main/java/com/zoomcar/uikit/disclaimer/ZDisclaimerView.kt [hunk 3]
```diff
@@ -48,34 +52,70 @@ class ZDisclaimerView : ConstraintLayout {
...
+                    R.style.CaptionTintedSunriseYellow
+                )
+                TextViewCompat.setTextAppearance(
+                    binding.textDisclaimer,
+                    R.style.CaptionTintedSunriseYellow
+                )
+                binding.disclaimerContainer.background = ContextCompat.getDrawable(
+                    context,
+                    R.drawable.background_sunrise_yellow_corner_radius_4dp
+                )
```
Comment (shishir5) [inline zoomdls/src/main/java/com/zoomcar/uikit/disclaimer/ZDisclaimerView.kt:104]: lets have common method setDisclaimerUiStyle and pass styles for textDisclaimierTitle, textDisclaimer and disclaimerContainer.background

File: zoomdls/src/main/java/com/zoomcar/uikit/disclaimer/ZDisclaimerView.kt [hunk 4]
```diff
@@ -86,13 +126,35 @@ class ZDisclaimerView : ConstraintLayout {
...
+            }
+        }
+    }
+
+    private fun getColorFromType(type: DisclaimerType): Int {
+        return when (type) {
+            DisclaimerType.INFO -> R.color.midnight_blue_06
+            DisclaimerType.WARNING -> R.color.sunrise_yellow_06
+            DisclaimerType.ERROR -> R.color.fire_red_06
+            else -> R.color.phantom_grey_10
```
Comment (shishir5) [inline zoomdls/src/main/java/com/zoomcar/uikit/disclaimer/ZDisclaimerView.kt:148]: what about success type??

**Review by shishir5 — CHANGES_REQUESTED** [review]

**Review by gideon-paul — COMMENTED** [review]
Umm. Why is .DSStore file included?

File: zoomdls/src/main/java/com/zoomcar/uikit/segmentedcontrol/ZSegmentedControl.kt [hunk 5]
```diff
@@ -75,8 +84,29 @@ class ZSegmentedControl : ConstraintLayout, ZSegmentedControlAdapter.IZSegmented
             isVisible = model?.header.isValid()
             measure(MeasureSpec.UNSPECIFIED, MeasureSpec.UNSPECIFIED)
         }
+
+        when (item.type) {
+            SegmentControlType.LABEL -> {
+                binding.segmentedControl.apply {
+                    setPadding(4, 4, 4, 4)
```
Comment (VaiSandeep) [inline zoomdls/src/main/java/com/zoomcar/uikit/segmentedcontrol/ZSegmentedControl.kt:91]: setpadding(4) also works

File: zoomdls/src/main/java/com/zoomcar/uikit/segmentedcontrol/ZSegmentedControl.kt [hunk 6]
```diff
@@ -75,8 +84,29 @@ class ZSegmentedControl : ConstraintLayout, ZSegmentedControlAdapter.IZSegmented
...
+                    background =
+                        ContextCompat.getDrawable(context, R.drawable.segmented_control_layout_bg)
+                }
+                layoutParams.height = UiUtil.dpToPixels(44, context)
+            }
+            SegmentControlType.ICON_LABEL -> {
+                binding.segmentedControl.apply {
+                    setPadding(0, 0, 0, 0)
```
Comment (VaiSandeep) [inline zoomdls/src/main/java/com/zoomcar/uikit/segmentedcontrol/ZSegmentedControl.kt:99]: same

**Review by VaiSandeep — APPROVED** [review]

**Review by rahulzoomcar — APPROVED** [review]
LGTM

**Review by gideon-paul — APPROVED** [review]
LGTM

**Review by nirajkulalzoomcar — APPROVED** [review]

**Review by shishir5 — APPROVED** [review]
LGTM

**Pull request merged**